File: java-jpa/src/main/java/member/Member.java
package member;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.time.LocalDateTime;
import lombok.Data;
import org.hibernate.annotations.DynamicInsert;

@Data
@Entity
@DynamicInsert
public class Member {
    @Id
    private String userId;
    private String password;
    private String email;
    private String role;
    private String tel;
    private Boolean activated;
    private LocalDateTime createdAt;
    private LocalDateTime rentableDate;
}

File: user-service/src/main/kotlin/com/grepp/app/model/member/entity/Member.kt
package com.grepp.spring.app.model.member.entity

import com.grepp.infra.entity.BaseEntity
import com.grepp.spring.app.model.auth.code.Role
import com.grepp.spring.infra.entity.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
class Member : BaseEntity() {
    @Id
    private val userId: String? = null
    private val password: String? = null
    private val email: String? = null

    @Enumerated(EnumType.STRING)
    private val role: Role? = null
    private val tel: String? = null

    @OneToOne(cascade = [CascadeType.PERSIST, CascadeType.MERGE])
    @JoinColumn(name = "userId")
    private val info: MemberInfo? = null

    fun updateLoginedAt(time: LocalDateTime?) {
        info.setLoginDate(time)
    }
}